pkg/operator/controllers/gameserver_controller.go

```diff
@@ -101,9 +101,12 @@ func (r *GameServerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if containsString(gs.GetFinalizers(), finalizerName) {
 			patch := client.MergeFrom(gs.DeepCopy())
 			// our finalizer is present, so lets handle any external dependency
-			r.unassignPorts(&gs)
+			err := r.unassignPorts(&gs)
 			// remove our finalizer from the list and update it.
 			controllerutil.RemoveFinalizer(&gs, finalizerName)
```

Comment:
Nit: check the value of `err` immediately after line 104 before calling RemoveFinalizer, though it should no impact.